File: cli-clients/java-client/CrudGeneratorCLI.java
package com.basiccode.cli;

import picocli.CommandLine;
import picocli.CommandLine.Command;
import picocli.CommandLine.Option;
import picocli.CommandLine.Parameters;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.concurrent.Callable;

@Command(name = "crud-generator", 
         mixinStandardHelpOptions = true,
         version = "1.0.0",
         description = "UML-to-CRUD Generator Java CLI Client")
public class CrudGeneratorCLI implements Callable<Integer> {
    
    @Parameters(index = "0", description = "Command: generate, validate")
    private String command;
    
    @Option(names = {"-u", "--uml"}, required = true, description = "UML diagram file")
    private Path umlFile;
    
    @Option(names = {"-o", "--output"}, required = true, description = "Output directory")
    private String outputPath;
    
    @Option(names = {"-f", "--framework"}, description = "Target framework", defaultValue = "spring-boot")
    private String framework;
    
    @Option(names = {"-p", "--package"}, description = "Package name", defaultValue = "com.example")
    private String packageName;
    
    @Option(names = {"--api-url"}, description = "API URL", defaultValue = "http://localhost:8080")
    private String apiUrl;
    
    @Override
    public Integer call() throws Exception {
        System.out.println("🚀 UML-to-CRUD Generator (Java CLI)");
        
        switch (command.toLowerCase()) {
            case "generate" -> generateCode();
            case "validate" -> validateUml();
            default -> {
                System.err.println("❌ Unknown command: " + command);
                return 1;
            }
        }
        
        return 0;
    }
    
    private void generateCode() throws IOException, InterruptedException {
        System.out.println("📁 Generating " + framework + " project from " + umlFile);
        
        String umlContent = Files.readString(umlFile);
        
        String jsonPayload = String.format("""
            {
                "umlContent": "%s",
                "packageName": "%s",
                "outputPath": "%s",
                "generationType": "COMPLETE_PROJECT",
                "framework": "%s"
            }
            """, 
            umlContent.replace("\"", "\\\"").replace("\n", "\\n"),
            packageName,
            outputPath,
            framework.toUpperCase().replace("-", "_")
        );
        
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(apiUrl + "/api/v2/generate/files"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonPayload))
            .build();
        
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        
        if (response.statusCode() == 200) {
            System.out.println("✅ Generation completed successfully!");
            System.out.println("📂 Project created at: " + outputPath);
            showNextSteps();
        } else {
            System.err.println("❌ API Error: " + response.statusCode());
            System.err.println(response.body());
        }
    }
    
    private void validateUml() throws IOException, InterruptedException {
        System.out.println("🔍 Validating " + umlFile);
        
        String umlContent = Files.readString(umlFile);
        
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(apiUrl + "/api/generate/validate"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString("\"" + umlContent.replace("\"", "\\\"") + "\""))
            .build();
        
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        
        if (response.statusCode() == 200) {
            System.out.println("✅ UML validation completed");
        } else {
            System.err.println("❌ Validation failed");
        }
    }
    
    private void showNextSteps() {
        System.out.println("\n🎯 Next Steps:");
        switch (framework) {
            case "spring-boot" -> {
                System.out.println("  cd " + outputPath);
                System.out.println("  mvn spring-boot:run");
            }
            case "django" -> {
                System.out.println("  cd " + outputPath);
                System.out.println("  pip install -r requirements.txt");
                System.out.println("  python manage.py runserver");
            }
        }
    }
    
    public static void main(String[] args) {
        int exitCode = new CommandLine(new CrudGeneratorCLI()).execute(args);
        System.exit(exitCode);
    }
}

File: cli-clients/python-client/crud_generator_cli.py
#!/usr/bin/env python3
"""
CLI Client pour UML-to-CRUD Generator
Usage: python crud_generator_cli.py generate --uml diagram.mermaid --output ./my-project --framework spring-boot
"""

import click
import requests
import json
import os
from pathlib import Path

API_BASE_URL = "http://localhost:8080/api/v2/generate"

@click.group()
@click.version_option(version="1.0.0")
def cli():
    """UML-to-CRUD Generator CLI Client"""
    pass

@cli.command()
@click.option('--uml', '-u', required=True, type=click.Path(exists=True), help='UML diagram file (.mermaid)')
@click.option('--output', '-o', required=True, type=click.Path(), help='Output directory path')
@click.option('--package', '-p', default='com.example', help='Base package name')
@click.option('--framework', '-f', 
              type=click.Choice(['spring-boot', 'django', 'flask', 'dotnet', 'express']),
              default='spring-boot', help='Target framework')
@click.option('--type', '-t',
              type=click.Choice(['files-only', 'complete-project']),
              default='complete-project', help='Generation type')
@click.option('--api-url', default=API_BASE_URL, help='API base URL')
def generate(uml: str, output: str, package: str, framework: str, type: str, api_url: str):
    """Generate CRUD code from UML diagram"""
    
    click.echo(f"🚀 Generating {framework} project from {uml}")
    click.echo(f"📁 Output directory: {output}")
    
    # Read UML content
    with open(uml, 'r') as f:
        uml_content = f.read()
    
    # Prepare request
    payload = {
        "umlContent": uml_content,
        "packageName": package,
        "outputPath": os.path.abspath(output),
        "generationType": type.upper().replace('-', '_'),
        "framework": framework.upper().replace('-', '_')
    }
    
    try:
        response = requests.post(f"{api_url}/files", json=payload, timeout=30)
        
        if response.status_code == 200:
            result = response.json()
            if result['success']:
                click.echo(f"✅ {result['message']}")
                click.echo(f"📊 Generated {result['classCount']} classes")
                show_next_steps(framework, output)
            else:
                click.echo(f"❌ Generation failed: {result['message']}", err=True)
        else:
            click.echo(f"❌ API Error: {response.status_code}", err=True)
            
    except requests.exceptions.RequestException as e:
        click.echo(f"❌ Connection error: {e}", err=True)

def show_next_steps(framework: str, output_path: str):
    """Show framework-specific next steps"""
    
    click.echo("\n🎯 Next Steps:")
    
    if framework == 'spring-boot':
        click.echo("  cd " + output_path)
        click.echo("  mvn spring-boot:run")
        
    elif framework == 'django':
        click.echo("  cd " + output_path)
        click.echo("  pip install -r requirements.txt")
        click.echo("  python manage.py runserver")

if __name__ == '__main__':
    cli()